### PR TITLE
Add kube config secret

### PR DESCRIPTION
**Pull request by carolineechen**

### CODE REVIEW

**Comment by carolineechen** [comment]
* **#1346** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1346?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @carolineechen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/secrets/provider_secrets/kubeconfig_secret.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+from runhouse.utils import create_local_dir
+
+
+class KubeConfigSecret(ProviderSecret):
+    """
+    .. note::
+        To create a KubeConfigSecret, please use the factory method :func:`provider_secret` with ``provider=="kubernetes"``.
+    """
+
+    _PROVIDER = "kubernetes"
```
Comment (carolineechen) [inline runhouse/resources/secrets/provider_secrets/kubeconfig_secret.py:18]: I named the class kube config to be more explicit, but should we also have the provider here be `"kubeconfig"` or leave as `"kubernetes"`? the flow to construct the secret from factory methods would be `rh.provider_secret("kubernetes/kubeconfig")`

File: tests/fixtures/secret_fixtures.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
     "lambda": {"api_key": "test_api_key"},
     "github": {"oauth_token": "test_oauth_token"},
     "huggingface": {"token": "test_token"},
+    "kubernetes": {"cluster": [{"cluster": "data"}], "context": {}},
```
[the remark this answers is not in the document]
Comment (carolineechen) [reply tests/fixtures/secret_fixtures.py]: updated the testing to reflect the actual fields in the structure better, the `secret.values` basically just stores the dict representation of the config. lmk if that works or if another format makes more sense

``` 
# secret.values
{
    "apiVersion": "v1",
    "clusters": {
        ["cluster": {...}]
    },
    "contexts": {
        ["context": {}]
    },
    "current-context": "xxx",
   ...
}
```

File: runhouse/resources/secrets/provider_secrets/kubeconfig_secret.py (hunk 1 above)
Comment (jlewitt1) [reply runhouse/resources/secrets/provider_secrets/kubeconfig_secret.py:18]: @dongreenberg you have a strong opinion about this? thinking ahead would there be other k8s data we might want to store?

**Review by jlewitt1 — APPROVED** [review]

**Pull request merged**